r/inst/build_arrow_static.sh

```diff
@@ -74,6 +74,8 @@ ${CMAKE} -DARROW_BOOST_USE_SHARED=OFF \
     -DARROW_DATASET=${ARROW_DATASET:-ON} \
     -DARROW_DEPENDENCY_SOURCE=${ARROW_DEPENDENCY_SOURCE:-AUTO} \
     -DAWSSDK_SOURCE=${AWSSDK_SOURCE:-} \
+    -DBoost_SOURCE=${BOOST_SOURCE:-} \
+    -Dlz4_SOURCE=${LZ4_SOURCE:-} \
```

Comment:
How about using same name for CMake variable? (We did so for `re2_SOURCE`/`xsimd_SOURCE`/`zstd_SOURCE`.)

```suggestion
    -DBoost_SOURCE=${Boost_SOURCE:-} \
    -Dlz4_SOURCE=${lz4_SOURCE:-} \
```

Reply:
Happy to change these — they are currently all caps due to the comment above this one: https://github.com/apache/arrow/pull/39587/files#r1450998393

We have separate code that is demanding all env vars are supposed to be all caps, so I just fit into that paradigm (and for a smaller patch) but I'm happy to remove that here and now.

Reply:
I see.

But it (all caps) may not work for existing `re2_SOURCE`/`xsimd_SOURCE`/`zstd_SOURCE`: https://github.com/apache/arrow/blob/87ed8ac432dc7e77fa0275a24e3edda9b0bc2a2e/r/inst/build_arrow_static.sh#L100-L102

(Sorry. I haven't execute these scripts. I just read the existing codes.)

Reply:
_nods_ yeah, those should have been untouched in this case, but regardless I've taken your suggestion here and made these match the cmake args through out. Thanks!

Reply:
This is done now. Thank you again for the suggestion